geps/gep-1713/index.md

```diff
@@ -0,0 +1,381 @@
+
+#### Addresses
+
+The list of `Addresses` that appear in the status of the "child" Gateway MUST be the same as the "parent" Gateway.
+
+#### Gateway Conditions
+
+Gateway conditions currently supports the following condition types: `Accepted` and `Programmed`
+
+For parent gateways, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a child listener is invalid, `ListenersNotValid` would be reported.
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
+
+For child gateways, `Accepted` and `Programmed` should not consider the overall merged Gateway status, but only the child's own listeners.
+The exception to this is if the parent Gateway is not valid in any regard.
+
+For example, if I have a `parent`, `child-1`, and `child-2`:
+* If `parent` is entirely invalid (for example, an invalid `address`), all three Gateways will reported `Accepted=False`.
+* If `child-1` has an invalid listener, `parent` and `child-1` will report `ListenersNotValid`, while `child-2` will not.
+* If `child-1` references a parent that doesn't exist then `child-1` will report `Accepted=False`
+* If `child-1` references a parent that doesn't allow merging then `child-1` will report `Accepted=False`
+* If `child-1` references another child (eg. `child-2`) then `child-1` will report `Accepted=False` 
+* If `child-1` references itself then `child-1` will report `Accepted=False`
+* If `child-1` and `parent` have different gatewayClassNames then `child-1` will report `Accepted=False`
+* If `child-1` references a parent whose allowed child namespaces do not include `child-1`'s namespace, then `child-1` will report `Accepted=False`
+
+When reporting status of a child, an implementation SHOULD be cautious about what information from the parent or siblings are reported
+to avoid accidentally leaking sensitive information that the child would not otherwise have access to.
+
+#### Listener Conditions
+
+Listener conditions should only be set for listeners directly defined in a given Gateway.
+Parent gateways MUST NOT have children's resources in their listener conditions list.
+Children gateways MUST NOT have parent's or sibling's resources in their listener conditions list.
+An implementation MAY reject Listeners with ListenerConditionAccepted=False and Reason "TooManyListeners"
+
+#### Policy attachment
+
+Policies attached to a parent Gateway apply to both the parent and all children listeners.
+
+Policies that attach to a child Gateway apply to all listeners defined in that Gateway, but do not impact
+parent or sibling listeners.
+If the implementation cannot apply the policy to only specific listeners, it should reject the policy.
+
+## Future Goals
+
+### Requirement Level
+
+We want to keep this API very simple so that the merging requirement level could increase from `MAY` to `MUST`
+
+## Alternatives
```

Comment:
> Move more of this logic into Routes since those already support ~infinite merging with Gateways

I was under the impression this was off the table - but if it's _on_ the table that could simplify Knative's use case tremendously.

Reply:
>I was under the impression this was off the table

IMO Gateway merging would add an additional layer of hierarchy to the API. Given the complexity that entails, we really need to be open to other similarly complex approaches as well. Would love to see these alternatives written out a bit more to see if any of them can solve the use cases we've defined in this GEP. If so, we might find a solution that is at least somewhat simpler than Gateway merging.

Reply:
The abandoned effort for "route delegation" can be found at https://github.com/kubernetes-sigs/gateway-api/issues/1058 and cross-referenced links if there's any content you want to review or pull from there.

Reply:
Added more alternatives